example/generated-src/objc/TXSItemList.mm

```diff
@@ -21,7 +21,14 @@ + (nonnull instancetype)itemListWithItems:(nonnull NSArray<NSString *> *)items
 
 - (NSString *)description
 {
-    return [NSString stringWithFormat:@"<%@ %p items:%@>", self.class, self, self.items];
+    return [NSString stringWithFormat:@"<class, %@ : %p, dict, %@>", self.class, self, [[self toDict] description]];
+}
+
+- (NSDictionary *)toDict
+{
+    #define _djinni_hide_null_(_o_) ((_o_)?(_o_):([NSNull null]))
```

Comment:
This was discussed elsewhere, though I can't see the discussion when scrolling around now.  I think this use of a macro is really awkward.  I support the alternative of having the generator simply generate the required code for each field, using ?: directly for optional fields, and not for non-optional fields.  In this case, the "items" field isn't optional, so there shouldn't be any need to check for null anyway.